src/psyclone/tests/psyad/main_test.py

```diff
@@ -624,7 +624,20 @@ def test_main_otest_lfric(tmpdir, capsys):
           "-otest", harness_out])
     output, error = capsys.readouterr()
     assert error == ""
-    assert output == ""
+    # TODO 2668: This deprecation of options causes this test to see the
+    # deprecation message.
+    correct = '''PSyclone Deprecation Warning: The options parameters to \
+Transformation.apply and Transformation.validate are \
+now deprecated. Please use the individual \
+arguments, or unpack the options with **options. \
+See the documentation for more details.
+PSyclone Deprecation Warning: The options parameters to \
```

Comment:
Seeing this message twice makes me wonder - is it because we're generating it from both `apply` and `validate`? In which case we should probably only do it in `validate` as `apply` should always call that. The other possibility is that psyad uses >1 transformation (which I'm pretty sure it does).

Reply:
I think this was coming from validate twice - but yeah validate only is probably sufficient.

Reply:
Apparently it was coming from apply twice - I guess because something is calling Transformation.apply but not Transformation.validate - not sure where this comes from.